bootstrap/util/configowner_test.go

```diff
@@ -20,113 +20,134 @@ import (
 	"context"
 	"testing"
 
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/utils/pointer"
-	"sigs.k8s.io/controller-runtime/pkg/client/fake"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-func TestGetConfigOwnerSuccess(t *testing.T) {
+func TestGetConfigOwner(t *testing.T) {
+	g := NewGomegaWithT(t)
+
 	scheme := runtime.NewScheme()
-	if err := clusterv1.AddToScheme(scheme); err != nil {
-		t.Fatal("failed to register Cluster API objects to scheme")
-	}
+	g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
 
-	myMachine := &clusterv1.Machine{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "my-machine",
-			Namespace: "my-ns",
-			Labels: map[string]string{
-				clusterv1.MachineControlPlaneLabelName: "",
+	t.Run("should get the owner when present (Machine)", func(t *testing.T) {
```

Comment:
Each of these `t.Run` should probably also have a `g := NewWithT(t)` to avoid issues with parallelism and the shared `g` above.

Reply:
Please resolve this issue, then lgtm

Reply:
Does this need to be resolved now? We don't run the tests in parallel as far as I know, and this is how we write tests elsewhere as well, we should revisit once we go back and look at our tests

Reply:
I think we should address it now, The `testing.T` that is passed into the subtest is not the same one that we invoke `NewWithT` with above: https://golang.org/src/testing/testing.go?s=32873:32920#L994

Reply:
done, thanks